lib/generators/deimos/schema_class_generator.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+
+require 'rails/generators'
+require 'deimos'
+require 'deimos/utils/schema_class_mixin'
+require 'deimos/config/configuration'
+
+# Generates new schema classes.
+module Deimos
+  module Generators
+    # Generator for Schema Classes used for the IDE and consumer/producer interfaces
+    class SchemaClassGenerator < Rails::Generators::Base
+      include Deimos::Utils::SchemaClassMixin
+
+      SPECIAL_TYPES = %i(record enum).freeze
+
+      source_root File.expand_path('schema_class/templates', __dir__)
+
+      argument :full_schema,
+               desc: 'The fully qualified schema name or path.',
+               required: false,
+               type: :string
+
+      no_commands do
+        # Retrieve the fields from this Avro Schema
+        # @return [Array<SchemaField>]
+        def fields
+          @current_schema.fields.map { |field| Deimos::SchemaField.new(field.name, field.type) }
+        end
+
+        # Converts Deimos::SchemaField's to String form for generated YARD docs
+        # @param schema_field [Deimos::SchemaField]
+        # @return [String] A string representation of the Type of this SchemaField
+        def deimos_field_type(schema_field)
+          _field_type(schema_field.type)
+        end
+
+        # @param enum [Avro::Schema::EnumSchema] a field of type 'enum'.
+        # @return [Array<String>] of symbols valid for the enum.
+        def enum_symbols(enum)
+          enum.symbols
+        end
```

Comment:
Seems a bit redundant? :)

Reply:
lol thats silly, i _think_ matthew left it over, will clean up this and other ones haha